### PR TITLE
Shelby/breakup autobuild

### PR DESCRIPTION
**Pull request by sfsholden**
### What does this PR do?
In order to reduce build time, tests have been split up into multiple jobs. These jobs will run in parallel if both VMs are free. Also in order to address test results being piped to the same file for the package, each test result will now output to its own junit-custom and xunit.xml file. Time rollups have been added to each piece of the appveyor build process for easier research going forward.

### What issues does this PR fix or reference?
@W-4889431@

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -50,18 +63,21 @@ test_script:
...
   - ps: |
-      npm run coverage:system-tests
       npm run aggregateJUnit
       junit-merge -d junit-aggregate -o junit-aggregate.xml
       $NpmTestsExitCode = $LastExitCode
       # Show status
       "`n`n"
       "Finalizing results`n"
-      $AllFiles = Get-ChildItem -Path $ProjectRoot\packages\*\xunit.xml -File | Select -ExpandProperty FullName
+      $AllFiles = Get-ChildItem -Path $ProjectRoot\packages\*\xunit*.xml -File | Select -ExpandProperty FullName
```
Comment (sfsholden) [inline .appveyor.yml]: Modified path to check for the new xunit variations (unit, integration, vscodeIntegration).

File: .appveyor.yml [hunk 2]
```diff
@@ -17,10 +17,11 @@ environment:
   SFDX_AUTOUPDATE_DISABLE: true
   SFDX_DOMAIN_RETRY: 300
   matrix:
-    - vscode: test:without-system-tests
-    #- vscode: test:insiders-without-system-tests
+    - vscode: test:unit-and-system-tests
+    - vscode: test:integration-tests-without-system-tests
```
Comment (sfsholden) [inline .appveyor.yml]: Job1 = unit tests and system tests; Job2 = integration tests and vscode integration tests.

File: config/mochaIntegrationTestsConfig.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "xunitReporterOptions": {
+    "output": "xunit-integrationTests.xml"
+  },
+  "mochaJunitReporterReporterOptions": {
+    "mochaFile": "junit-custom-integrationTests.xml"
+  },
+  "colors": true,
+  "slow": 0,
+  "fullTrace": true
+}
```
Comment (sfsholden) [inline config/mochaIntegrationTestsConfig.json]: New MochaConfig spec to run for integration tests.

File: config/mochaUnitTestsConfig.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
 {
   "reporterEnabled": "mocha-junit-reporter, xunit, spec",
   "xunitReporterOptions": {
-    "output": "xunit.xml"
+    "output": "xunit-unitTests.xml"
   },
   "mochaJunitReporterReporterOptions": {
-    "mochaFile": "junit-custom.xml"
+    "mochaFile": "junit-custom-unitTests.xml"
```
Comment (sfsholden) [inline config/mochaUnitTestsConfig.json]: Modified MochaConfig to output results for unit tests.

File: scripts/aggregate-junit-xml.js [hunk 5]
```diff
@@ -3,24 +3,28 @@ const fs = require('fs-extra');
...
+dirs.filter(dir => {
+  junitFilesToCheck.forEach(function(junitFile) {
+    var fullFilePath = path.join(packagesDir, dir, junitFile);
+    if (fs.existsSync(fullFilePath)) {
+      shell.cp(
+        fullFilePath,
+        path.join(process.cwd(), `junit-aggregate/${dir}-${junitFile}`)
+      );
+    }
+  });
```
Comment (sfsholden) [inline scripts/aggregate-junit-xml.js]: For each package directory, check to see if the junit-custom* files are present. If the file is present then copy it to the junit-aggregate folder and prepend the package directory to the file name.

**Comment by lcampos** [comment]
@sfsholden these changes look good, are we also updating the timeout value in appveyor to be lower than 60 mins per job ?

**Comment by sfsholden** [comment]
@lcampos Yes, I would still like to change the build timeout limit. I'm thinking 35 minutes max for each job for now. I was planning on adding this in the Appveyor UI Settings page. There is an option for setting 'Build timeout, minutes'. Once this is merged with develop, I would like to change this setting. Let me know what you think. Thanks!

**Comment by lcampos** [comment]
@sfsholden that sounds good, let's update the branch and get it merged

**Pull request merged**